zingo-sync/src/scan.rs

```diff
@@ -524,3 +545,143 @@ fn collect_nullifiers(
         });
     Ok(())
 }
+
+async fn scan_transactions<P>(
+    fetch_request_sender: mpsc::UnboundedSender<FetchRequest>,
+    parameters: &P,
+    scanning_keys: &ScanningKeys,
+    relevant_txids: HashSet<TxId>,
+    decrypted_note_data: DecryptedNoteData,
+    wallet_blocks: &BTreeMap<BlockHeight, WalletBlock>,
+) -> Result<HashMap<TxId, WalletTransaction>, ()>
+where
+    P: Parameters,
+{
+    let mut wallet_transactions = HashMap::with_capacity(relevant_txids.len());
```

Comment:
because a txid may include multiple transactions, this here must be a HashSet rather than a Hashmap

Reply:
a txid is a unique id for a single transaction. wallet_transactions is the equivalent of transaction_records_by_id in zingolib which is also a hashmap

Reply:
just so you are aware hashsets also are unique lists with no duplicates